azkaban-common/src/main/java/azkaban/executor/container/ContainerizedDispatchManager.java

```diff
@@ -174,13 +178,33 @@ public DispatchMethod getDispatchMethod(final ExecutableFlow flow) {
   }
 
   /**
-   * This method is used to start queue processor thread. The queue processor thread will pick
-   * execution from queue maintained in database and create a container for it.
+   * This is responsible for starting processing loops for the containerized dispatch.
+   * Currently this includes:
+   * (1) This method is used to start queue processor thread. The queue processor thread will pick
+   *   execution from queue maintained in database and create a container for it.
+   * (2) Starting container event watch, if present.
    */
   @Override
   public void start() {
     this.queueProcessor = setupQueueProcessor();
     this.queueProcessor.start();
+    this.startWatch();
+  }
+
+  // Start the event watch if configured.
+  private void startWatch() {
+    if (!containerizedWatch.isPresent()) {
```

Comment:
if (!containerizedWatch.isPresent() || !this.azkProps.getBoolean(ContainerizedDispatchManagerProperties.KUBERNETES_WATCH_ENABLED,
        false)) {
log
return;
}

Do you want to reduce cases by adding false check for property in if case?

Reply:
The logs in the two separate `if` blocks convey slightly different information. In one case the watch is not initialized and in the other it's explicitly disabled because of the property. I would prefer to keep the separate as it will make it easier to debug future issues from the log file. 
On a sidenote, this kind of a layout would have been avoidable with the use of `OptionalBinder` in guice, but it wasn't clear if the version we use fully supported it. We should consider updating guice in the near future.